### PR TITLE
chore(www): convert HomepageBlogPosts to function component

### PR DESCRIPTION
**Pull request by abhishekjakhar**
## Description

Convert the `<HomepageBlogPosts/>` to function component from class based component

**Note:** There is no change functionality wise, just the implementation is different.

### How to test

1. Go to homepage of Gatsby using the links provided below
2. Scroll down to the **The Gatsby blog** section on homepage(screenshots are provided below)
3. This should work same as before
4. The layout of blog posts should change when resizing the browser to mobile screen sizes

**Local URL:** http://localhost:8000/
**Production URL:** https://www.gatsbyjs.org/

#### Desktop
homepage-blog-posts-desktop

#### Mobile
homepage-blog-posts-mobile

### CODE REVIEW

File: www/src/components/homepage/homepage-blog-posts.js [hunk 1]
```diff
@@ -114,28 +114,22 @@ const ViewAll = () => (
...
+  }, [])
 
-  updateViewPortState = () => {
-    this.setState({ desktopViewport: this.desktopMediaQuery.matches })
+  const updateViewPortState = () => {
+    setDesktopViewport(desktopMediaQuery.matches)
   }
 
-  splitPostsToColumns = posts =>
+  const splitPostsToColumns = posts =>
```
Comment (tesseralis) [inline www/src/components/homepage/homepage-blog-posts.js:132]: You can pull this out of the component since it doesn't have any references to props/state

**Comment by abhishekjakhar** [comment]
@fk can I get a review on this PR?

File: www/src/components/homepage/homepage-blog-posts.js [hunk 2]
```diff
@@ -114,108 +114,96 @@ const ViewAll = () => (
...
+  useEffect(() => {
+    desktopMediaQuery = window.matchMedia(`(min-width: ${breakpoints[3]}`)
+    desktopMediaQuery.addListener(updateViewPortState)
+    setDesktopViewport(desktopMediaQuery.matches)
+    return () => desktopMediaQuery.removeListener(updateViewPortState)
+  }, [])
 
-  updateViewPortState = () => {
-    this.setState({ desktopViewport: this.desktopMediaQuery.matches })
+  const updateViewPortState = () => {
```
Comment (tesseralis) [inline www/src/components/homepage/homepage-blog-posts.js:142]: Move this definition inside the `useEffect`.

File: www/src/components/homepage/homepage-blog-posts.js [hunk 3]
```diff
@@ -114,108 +114,96 @@ const ViewAll = () => (
...
-  }
+export default function HomepageBlogPosts({ posts }) {
+  const [desktopViewport, setDesktopViewport] = useState(false)
+  let desktopMediaQuery = null
 
-  componentWillUnmount = () => {
-    this.desktopMediaQuery.removeListener(this.updateViewPortState)
-  }
+  useEffect(() => {
+    desktopMediaQuery = window.matchMedia(`(min-width: ${breakpoints[3]}`)
```
Comment (tesseralis) [inline www/src/components/homepage/homepage-blog-posts.js:136]: Delete the definition above and do this:

```suggestion
    const desktopMediaQuery = window.matchMedia(`(min-width: ${breakpoints[3]}`)
```

File: www/src/components/homepage/homepage-blog-posts.js [hunk 4]
```diff
@@ -114,108 +114,93 @@ const ViewAll = () => (
...
+    [[], []]
+  )
+
+export default function HomepageBlogPosts({ posts }) {
+  const [desktopViewport, setDesktopViewport] = useState(false)
+
+  useEffect(() => {
+    const updateViewPortState = () =>
+      setDesktopViewport(desktopMediaQuery.matches)
+    const desktopMediaQuery = window.matchMedia(`(min-width: ${breakpoints[3]}`)
```
Comment (tesseralis) [inline www/src/components/homepage/homepage-blog-posts.js:137]: Ooh, I think we can replace this with `mediaQueries.lg` for clarity.

Comment (abhishekjakhar) [reply www/src/components/homepage/homepage-blog-posts.js:137]: `mediaQueries.lg` returns `@media (min-width: 1000px)`, but we are passing only `(min-width: 1000px)` to window.matchMedia. If we use `mediaQueries.lg` then we will pass extra @media to matchMedia.

**Comment by fk** [comment]
Thank y'all! 🙏

**Comment by abhishekjakhar** [comment]
> Thank y'all! 🙏

Thank you too 🙏

**Pull request merged**